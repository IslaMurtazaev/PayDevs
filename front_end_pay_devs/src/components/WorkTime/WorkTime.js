import React from "react";
import { Link } from "react-router-dom";

const WorkTimeItem = props => {
  const { projectId, hourPaymentId, workTime, onRemove } = props;
  return (
    <div className="workedTime">
      <h4><b>Start Work:</b> {new Date(workTime.start_work).toLocaleString()} </h4>
      <h4><b>End Work:</b> {new Date(workTime.end_work).toLocaleString()}</h4>
      <h4>{workTime.paid || "not"} <b>paid</b></h4>
      <div className="button-group-horizontal">
      <Link 
        to={`/project/${projectId}/Hourly/${hourPaymentId}/workTime/${
          workTime.id
        }/update`}  
      >
        <button className="updateButton btn btn-warning" type="button">Update</button>
      </Link>
      <button type="button" className="btn btn-danger" onClick={() => onRemove(workTime.id)}>Delete</button>
      </div>
         </div> 
  );
};

export default WorkTimeItem;
