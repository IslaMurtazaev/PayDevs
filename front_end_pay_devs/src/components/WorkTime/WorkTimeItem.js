import React, { Component } from "react";
// import {Link} from "react-router-dom";

class WorkTimeItem extends Component {

    onClickDelete(workTimeId){
        this.props.onDelete(workTimeId);

    }

    render() {
        const { workTime} = this.props;
        return (
            <div>
                <div>Start Work: {workTime.start_work}</div>
                <div>End Work: {workTime.end_work}</div>
                {workTime.paid ? <div>paid</div>: <div>not paid</div>}
                {/* <button><Link to={`/project/${projectId}/Hourly/${workTime.id}/update`}>UPDATE</Link></button> */}
                <button onClick={this.onClickDelete.bind(this, workTime.id)}>DELETE</button>
            </div>
        );
    }
}

export default WorkTimeItem;