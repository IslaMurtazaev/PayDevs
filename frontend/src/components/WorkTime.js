import React from "react";
import { Link } from "react-router-dom";

const WorkTimeItem = props => {
  const { projectId, hourPaymentId, workTime, onRemove } = props;

  return (
    <div className="workedTime">
      <h4 className="start_work">
        <b>Start Work:</b> {new Date(workTime.start_work).toLocaleString()}
      </h4>
      <h4 className="end_work">
        <b>End Work:</b> {new Date(workTime.end_work).toLocaleString()}
      </h4>
      <h4 className="paid">
        {workTime.paid || "not "}
        <b>paid</b>
      </h4>
      <div className="button-group-horizontal">
        <Link
          to={`/project/${projectId}/Hourly/${hourPaymentId}/workTime/${
            workTime.id
          }/update`}
          className="updateWorkTime"
        >
          <button className="updateButton btn btn-warning" type="button">
            Update
          </button>
        </Link>
        <button
          type="button"
          className="btn btn-danger removeWorkTime"
          onClick={() => onRemove(workTime.id)}
        >
          Delete
        </button>
      </div>
    </div>
  );
};

export default WorkTimeItem;
