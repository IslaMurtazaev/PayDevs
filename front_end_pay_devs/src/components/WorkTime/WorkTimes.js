import React, { Component } from "react";
import { connect } from "react-redux";
import {workTimeActions} from '../../actions/workTime';
import WorkTimeItem from "./WorkTimeItem"
import FormikWorkTime from "../../forms/FormikWorkTime";




class WorkTimes extends Component {
  onClick() {
    this.props.onGetAllProjects();
  }

  componentDidMount() {
    const {hourPayment } = this.props;
    this.props.onGetAll(hourPayment.id);
  }

  render() {
    
    const { hourPayment, projectId, workTimes} = this.props;
    return (
      <div>
        <ul>
          {workTimes.map(workTime=>(
            <li key={workTime.id}>
                <WorkTimeItem hourPayment={hourPayment}  
                workTime = {workTime}
                projectId={projectId} 
                onDelete={this.props.onDeletehourPayment}/>
            </li>)
          )
        }
        </ul>
        <FormikWorkTime onSubmit={this.props.createWorkTime} 
          projectId={projectId} 
          hourPaymentId = {this.props.hourPayment.id}
          /> 
          
      </div>
    );
  }
}

const mapStateToProps = (state) => {
    
    return {
      workTimes: state.workTimes
    };
  };

const mapDispatchToProps = (dispatch) => {
    return {
        onGetAll:  hourPaymentId => dispatch(workTimeActions.getAll(hourPaymentId)),
        onDeletehourPayment: (hourId) => {
            dispatch(workTimeActions.deleteWork(hourId))
        },
        createWorkTime: values => {
            dispatch(workTimeActions.create(values))
          },
    }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(WorkTimes);
