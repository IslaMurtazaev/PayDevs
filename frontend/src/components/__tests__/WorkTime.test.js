import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import WorkTime from "../WorkTime";

describe("<WorkTime />", () => {
  let workTime;
  beforeEach(() => {
    workTime = {
      id: 1,
      start_work: "2018-08-24T00:00:00.000Z",
      end_work: "2018-09-30T10:00:00.000Z",
      paid: true
    };
  });

  it("renders without errors", () => {
    shallow(<WorkTime workTime={workTime} />);
  });

  it("renders 'Invalid Date' if props.start(end)_work is not valid", () => {
    workTime.start_work = "02-08-1999T00:00:00.000Z";
    workTime.end_work = "05:01:10";
    const workTimeComponent = shallow(<WorkTime workTime={workTime} />);
    expect(workTimeComponent.find(".start_work").text()).toBe(
      "Start Work: Invalid Date"
    );
    expect(workTimeComponent.find(".end_work").text()).toBe(
      "End Work: Invalid Date"
    );
  });

  it("renders start_work based on props", () => {
    const workTimeComponent = shallow(<WorkTime workTime={workTime} />);
    expect(workTimeComponent.find(".start_work").text()).toBe(
      "Start Work: 2018-8-24 06:00:00"
    );
  });

  it("renders end_work based on props", () => {
    const workTimeComponent = shallow(<WorkTime workTime={workTime} />);
    expect(workTimeComponent.find(".end_work").text()).toBe(
      "End Work: 2018-9-30 16:00:00"
    );
  });

  it("renders paid based on props", () => {
    let workTimeComponent = shallow(<WorkTime workTime={workTime} />);
    expect(workTimeComponent.find(".paid").text()).toBe("paid");

    workTime.paid = false;
    workTimeComponent = shallow(<WorkTime workTime={workTime} />);
    expect(workTimeComponent.find(".paid").text()).toBe("not paid");
  });

  it("renders remove button based on props", () => {
    let onRemoveSpy = jest.fn();
    let workTimeComponent = shallow(
      <WorkTime workTime={workTime} onRemove={onRemoveSpy} />
    );
    workTimeComponent.find(".btn-danger").simulate("click");
    expect(onRemoveSpy).toBeCalledWith(workTime.id);
  });

  it("matches previous snapshot", () => {
    expect(toJson(shallow(<WorkTime workTime={workTime} />))).toMatchSnapshot();
  })
});
