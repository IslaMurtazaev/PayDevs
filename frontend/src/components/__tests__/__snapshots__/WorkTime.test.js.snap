// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkTime /> matches previous snapshot 1`] = `
<div
  className="workedTime"
>
  <h4
    className="start_work"
  >
    <b>
      Start Work:
    </b>
     
    2018-8-24 06:00:00
  </h4>
  <h4
    className="end_work"
  >
    <b>
      End Work:
    </b>
     
    2018-9-30 16:00:00
  </h4>
  <h4
    className="paid"
  >
    <Component />
    <b>
      paid
    </b>
  </h4>
  <div
    className="button-group-horizontal"
  >
    <Link
      className="updateWorkTime"
      replace={false}
      to="/project/undefined/Hourly/undefined/workTime/1/update"
    >
      <button
        className="updateButton btn btn-warning"
        type="button"
      >
        Update
      </button>
    </Link>
    <button
      className="btn btn-danger removeWorkTime"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;
